include/seqan3/search/algorithm/search_result_range.hpp

```diff
@@ -0,0 +1,331 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::search_result_range.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/range/views/single_pass_input.hpp>
+#include <seqan3/range/views/zip.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+
+namespace seqan3
+{
+
+/*!\brief An input range over the search results generated by the given search algorithm.
+ * \ingroup submodule_search_algorithm
+ * \implements std::ranges::input_range
+ *
+ * \tparam search_algorithm_t The type of the search algorithm; must model std::invocable with the value type of
+ *                            the given range query over queries.
+ * \tparam query_range_t The type of the range over the queries; must model std::ranges::forward_range.
+ *
+ * \details
+ *
+ * Provides a lazy input-range interface over the search results generated by the underlying search algorithm.
+ */
+template <typename search_algorithm_t, std::ranges::view query_range_t>
+//!\cond
+    requires std::ranges::forward_range<query_range_t> &&
+             std::invocable<search_algorithm_t, std::ranges::range_reference_t<query_range_t>>
+//!\endcond
+class search_result_range
+{
+private:
+    class search_result_range_iterator;
+
+    //!\brief Befriend the iterator type.
+    friend class search_result_range_iterator;
```

Comment:
isn't this enough?

Reply:
are these not two different expressions?